tests/integration/workspace_access/test_redash.py

```diff
@@ -0,0 +1,114 @@
+import json
+import logging
+from collections.abc import Iterable
+from datetime import timedelta
+
+from databricks.sdk.errors import NotFound
+from databricks.sdk.retries import retried
+from databricks.sdk.service import iam, sql
+
+from databricks.labs.ucx.workspace_access import redash
+from databricks.labs.ucx.workspace_access.base import Permissions
+from databricks.labs.ucx.workspace_access.groups import MigratedGroup, MigrationState
+from databricks.labs.ucx.workspace_access.redash import RedashPermissionsSupport
+
+from . import apply_tasks, apply_tasks_appliers, apply_tasks_crawlers
+
+logger = logging.getLogger(__name__)
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=3))
+def test_permissions_for_redash(
+    ws,
+    sql_backend,
+    inventory_schema,
+    make_ucx_group,
+    make_group,
+    make_user,
+    make_query,
+    make_query_policy_permissions,
+):
+    ws_group = make_group()
+    user = make_user()
+
+    query = make_query()
+    make_query_policy_permissions(
+        object_id=query.id,
+        permission_level=sql.PermissionLevel.CAN_EDIT,
+        group_name=ws_group.display_name,
+        user_name=user.display_name,
+    )
+
+    group_to_migrate = MigratedGroup.partial_info(ws_group, ws_group)
+    group_to_migrate.temporary_name = None  # don't apply permissions for the temp group
```

Comment:
This doesn't look correct - can you create two different groups, like all other similar tests? You're creating a single group and doing these weird things related to empty temporary name

Reply:
corrected